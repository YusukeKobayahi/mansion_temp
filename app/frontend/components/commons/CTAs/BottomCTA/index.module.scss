@import "~/styles/variables";
@import "~/styles/utils";

.container {
  max-width: 650px;
  background-color: unset;
  margin-top: 50px;
  padding: 0;
  border: 1px solid #e5e5e5;
  border-radius: 3px;

  @include screenMd {
    margin-top: 32px;
  }
}

.bottom {
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-between;

  @include screenMd {
    padding: 8px;
  }
}

.texts {
  text-align: left;
  font-size: 15px;
  color: #646464;
  margin-right: 5px;

  @include screenMd {
    font-size: 2vw;
  }
}

.btn_container {
  padding: 1rem;
}

.btn {
  width: 100%;
  color: #c33e32;
  line-height: 1;
  padding: 16px 35px 16px 16px;
  border-radius: 4px;
  border: 1px solid #c33e32;
  position: relative;
  font-size: 16px;
  white-space: nowrap;
  font-weight: bold;
  background-color: white;
  box-shadow: 1px 1px 4px grey;

  @include screenMd {
    font-size: 2.3vw;
    padding: 2.2vw 4.5vw 2.2vw 2.2vw;
    line-height: 1.2;
    text-align: left;
  }

  &::after {
    content: "ï¼ž";
    position: absolute;
    top: 50%;
    right: 10px;
    transform: translateY(-50%);

    @include screenMd {
      right: 5px;
    }
  }
}
