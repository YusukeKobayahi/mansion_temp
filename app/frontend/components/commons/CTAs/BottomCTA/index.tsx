import Image from "next/image";
import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";

const BottomCTA: React.FC = () => {
  const clickCTA = () => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_ichiran",
      label: "msl_banner",
      value: 1,
    });
  };
  return (
    <button className={styles.container} onClick={clickCTA}>
      <div className={styles.image}>
        <Image
          src={"/ms-library/images/bottom-banner.png"}
          width={650}
          height={167}
          alt={"未公開物件を不動産会社から紹介"}
        />
      </div>
      <div className={styles.bottom}>
        <p className={styles.texts}>
          未公開物件に1番に申し込みができる！
          <br />
          Housiiは複数の不動産会社からあなたにあった物件を提案するサービスです。良質な未公開物件に1番早く申し込むことができます。
        </p>
      </div>
      <div className={styles.btn_container}>
        <div className={styles.btn}>まずは無料会員登録で希望を伝えてみる</div>
      </div>
    </button>
  );
};

export default BottomCTA;
