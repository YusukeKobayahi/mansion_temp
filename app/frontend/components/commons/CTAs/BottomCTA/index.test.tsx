import { render } from "@testing-library/react";
import BottomCTA from ".";

describe("components/commons/CTAs/BottomCTA", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<BottomCTA />);
    expect(asFragment()).toMatchSnapshot();
  });
});
