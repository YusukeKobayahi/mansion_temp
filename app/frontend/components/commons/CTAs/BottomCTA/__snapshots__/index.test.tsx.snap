// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CTAs/BottomCTA 正常に表示される 1`] = `
<DocumentFragment>
  <button
    class="container"
  >
    <div
      class="image"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjUwIiBoZWlnaHQ9IjE2NyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="未公開物件を不動産会社から紹介"
          data-nimg="true"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="bottom"
    >
      <p
        class="texts"
      >
        未公開物件に1番に申し込みができる！
        <br />
        Housiiは複数の不動産会社からあなたにあった物件を提案するサービスです。良質な未公開物件に1番早く申し込むことができます。
      </p>
    </div>
    <div
      class="btn_container"
    >
      <div
        class="btn"
      >
        まずは無料会員登録で希望を伝えてみる
      </div>
    </div>
  </button>
</DocumentFragment>
`;
